src/random.cpp

```diff
@@ -43,15 +43,21 @@ static void RandFailure()
 
 static inline int64_t GetPerformanceCounter()
 {
-    int64_t nCounter = 0;
-#ifdef WIN32
-    QueryPerformanceCounter((LARGE_INTEGER*)&nCounter);
+#if defined(_MSC_VER)
+    return __rdtsc();
+#elif defined(__i386__)
+    uint64_t r;
+    __asm__ volatile ("rdtsc" : "=A"(r));
+    return r;
+#elif defined(__x86_64__) || defined(__amd64__)
+    uint64_t r1, r2;
+    __asm__ volatile ("rdtsc" : "=a"(r1), "=d"(r2));
+    return (r2 << 32) | r1;
```

Comment:
I added some comments, but I don't think this warrants a full explanation of extended asm syntax in comments.

> Also, I couldn't find a description on the constraints a or d, I assume they are just writing to the inline variables r1 and r2, but I found documentation on this to be difficult to find.

See https://gcc.gnu.org/onlinedocs/gcc-4.9.0/gcc/Machine-Constraints.html#Machine-Constraints

"A" is the ax:dx register pair, so on x86 we constrain the `r` variable to be 64 bits consisting of 32 bits in ax and 32 bits in dx. Since rdtsc always stores its result in ax/dx, we get the correct result in the C++ variable r. "a" and "d" just constrain to the (r)ax and (r)dx registers, which is where the individual parts of the rdtsc result are stored on x86_64.

> Also, is volatile necessary? It doesn't hurt I suppose?

It's there to limit the optimizations the compiler can apply to the assembly block. See https://gcc.gnu.org/onlinedocs/gcc/Extended-Asm.html#Volatile